### PR TITLE
Python 3 everywhere, PEP8

### PR DESCRIPTION
**Pull request by int-ua**
#### Summary

SUMMARY: Infrastructure "All .py scripts ported to Python 3 and checked with PEP8 on moderate settings"

#### Purpose of change

Moving away from Python 2 as it's no longer officially maintained for half a year now: https://www.python.org/doc/sunset-python-2/

#### Describe the solution

Applying 2to3 to all .py files. Applying `pep8 --ignore=E501,E731,W605,E722` suggestions. Reverting some changes later and a couple of manual fixes.

#### Testing

Pending / In progress.

Asking some authors to help:
` build-data/osx/dmgsettings.py` @remyroy
`build-scripts/get_all_mods.py` @jbytheway @BevapDin 
`lang/extract_json_strings.py` @anothersimulacrum @Qrox 
`lang/strip_line_numbers.py` @yobbobanana @VlasovVitaly 
`lang/unicode_check.py` ^^
`tools/copy_from.py` @mlangsdorf 
`tools/dialogue_validator.py` ^ 
`tools/merge_maps.py` ^
`tools/palettize.py` ^
`tools/gfx_tools/compose.py` ^
`tools/gfx_tools/decompose.py` ^
`tools/gfx_tools/png_update.py` ^
`tools/fix-compilation-database.py` Qrox
`tools/update_blueprint_needs.py` ^
`tools/generate_changelog.py` @KurzedMetal 
`tools/json_tools/cddatags.py` jbytheway
`tools/json_tools/values.py` @jeremyosborne
`tools/json_tools/keys.py` ^
`tools/json_tools/lister.py` ^ 
`tools/json_tools/pluck.py` ^ 
`tools/json_tools/splitter.py` ^
`tools/json_tools/util.py` ^
`tools/json_tools/table.py` wapcaplet
`tools/vehicleDef.py` @ymber @ralreegorganon
`tools/json_tools/adjust_values.py` ymber 
`tools/json_tools/pocket_mags.py` ^
`tools/windows_limit_memory.py` @neitsa 
`utilities/building-utility/deconstruct.py` @wormingdead 
`utilities/make_iso.py` @Leland

### CODE REVIEW

**Comment by wapcaplet** [comment]
Thanks for doing this. My quick-and-dirty unicode hack in the `table.py` script appears to be problematic in Py3:

```python
UTF8Writer = codecs.getwriter('utf8')
sys.stdout = UTF8Writer(sys.stdout)
```

Running `table.py --help` (or with any other argument) causes it to choke on line 168 `args = parser.parse_args()` with an `argument must be str, not bytes`. The above hack was taken from [unicode frustrations](https://pythonhosted.org/kitchen/unicode-frustrations.html); simply commenting it out allows the script to run again, but I haven't looked into alternatives yet. For now it may be OK to simply remove that unicode hack section.

I see several `print(...)` statements have been wrapped with extra `((...))` - maybe a global search/replace gone awry? I understand `print(...)` now requires parentheses but one set should be sufficient.

File: tools/json_tools/table.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def safe_value(value, format):
 
     """
     if format == 'md':
-        return value.replace('|', '\|')
+        return value.replace('|', '\\|')
```
Comment (int-ua) [inline tools/json_tools/table.py:64]: @wapcaplet Can you please take a look if PEP8 was wrong here?

Comment (wapcaplet) [reply tools/json_tools/table.py:59]: It appears to work either way, but the `\\|` version seems more correct to me.

>>> '|'.replace('|', '\|')
'\\|'
>>> '|'.replace('|', '\\|')
'\\|'

**Comment by jbytheway** [comment]
`get_all_mods` was already python 3. It looks like you've PEP-8-ified it, which is not a bad thing per se.

Since you're looking at tidying up the python scripts anyway, it would be nice if someone added some checking to the CI, like running flake8 across all of them. That probably shouldn't be in this PR, though.

**Comment by Qrox** [comment]
`lang/extract_json_strings.py` and `tools/fix-compilation-database.py` were already python 3 too.

**Comment by int-ua** [comment]
If you want some PEP8 changes reverted - I'm ok with that, just tell me.

File: lang/extract_json_strings.py [hunk 2]
```diff
@@ -1169,25 +1217,27 @@ def extract(item, infilename):
...
+        writestr(outfile, item["footsteps"], **kwargs)
+        wrote = True
+    if not wrote and "copy-from" not in item:
         if not warning_supressed(infilename):
             print("WARNING: {}: nothing translatable found in item: {}".format(infilename, item))
 
+
 def is_official_mod(full_path):
-    for i in official_mods:
+    for i in official_mods:  # FIXME: official_mods undefined
```
Comment (int-ua) [inline lang/extract_json_strings.py:1228]: @BrettDong Can you please help with that? Maybe you remember the context and can fix it quickly?

Comment (int-ua) [reply lang/extract_json_strings.py:1228]: Ok, looks like it was never used, I'm removing it.

**Comment by Qrox** [comment]
The PEP8 changes look good to me, thanks for doing it. And `tools/update_blueprint_needs.py` was already python 3 too, didn't notice it when I first read the OP.

File: lang/extract_json_strings.py (hunk 2 above)
Comment (BrettDong) [reply lang/extract_json_strings.py]: I have no idea of this function either.

File: tools/windows_limit_memory.py [hunk 3]
```diff
@@ -670,7 +670,7 @@ def main(args: argparse.Namespace) -> int:
         proc_limiter.assign_process_to_job()
         proc_limiter.limit_process_memory(args.memory)
         if not proc_limiter.wait_for_job():
-            input("Press <enter> to exit this script")
+            eval(input("Press <enter> to exit this script"))
```
Comment (int-ua) [inline tools/windows_limit_memory.py:673]: hmm

File: tools/gfx_tools/decompose.py [hunk 4]
```diff
@@ -304,15 +307,15 @@ def get_all_data(self, tileset_dirname, delete_pathname):
             os.stat(self.tileset_pathname)
         except KeyError:
             print("cannot find a directory {}".format(self.tileset_pathname))
-            exit -1
+            sys.exit(-1)
```
Comment (int-ua) [inline tools/gfx_tools/decompose.py:310]: @mlangsdorf It's 255 in the end, is it what you needed or should it be just `1`?

File: tools/windows_limit_memory.py (hunk 3 above)
Comment (neitsa) [reply tools/windows_limit_memory.py:673]: (script author) Yep, it's a full python 3 script, it work as it, as mentioned in the module docstring (and the shebang). No need to eval on `input`.

**Comment by int-ua** [comment]
Can I silence @lgtm-com ?

**Comment by int-ua** [comment]
Merged commits, reduced number of touched lines by not editing comments.

**Pull request merged**